### PR TITLE
Reading/Overriding configurations and Secrets from azure keyvault with MSI

### PR DESCRIPTION
**Pull request by manikarthikk**
#### What does this PR do? Please provide some context
Prerequisites: 
This change   has the dependency with [Azure MSI ](https://docs.microsoft.com/en-us/azure/active-directory/managed-identities-azure-resources/tutorial-linux-vm-access-arm)

This will read/override the configurations and secrets from azure key vault. 
Right now configs and secrets are read from lms.env.json&lms.auth.json and cms.env.json&cms.auth.json files. 
This change will take the secrets from azure keyvault and override the keys extracted from *.json files.  

we have to provide the following configs in json files (need to add in [configuration repo](https://github.com/edx/configuration/blob/master/playbooks/roles/edxapp/defaults/main.yml) ) to read from keyvault:

```
In lms.env.json: 

    "AZURE_MSI_REQUEST_URI" : "http://169.254.169.254/metadata/identity/oauth2/token",
    "GET_SECRETS_FROM_AZURE_KEYVAULT": true,
    "KEYVAULT_PAYLOAD":  {"resource": "https://vault.azure.net","api-version": "2018-02-01"},
    "KEYVAULT_URL" : "https://{keyvaultname}.vault.azure.net",
    "LMS_NON_SECURE_KEY_NAME": {key name from keyvault},
    "LMS_SECURE_KEY_NAME": {key name from keyvault },
    "API_VERSION": "2016-10-01"

In cms.env.json: 

    "AZURE_MSI_REQUEST_URI" : "http://169.254.169.254/metadata/identity/oauth2/token",
    "GET_SECRETS_FROM_AZURE_KEYVAULT": true,
    "KEYVAULT_PAYLOAD":  {"resource": "https://vault.azure.net","api-version": "2018-02-01"},
    "KEYVAULT_URL" : "https://{keyvaultname}.vault.azure.net",
    "CMS_NON_SECURE_KEY_NAME": {key name from keyvault},
    "CMS_SECURE_KEY_NAME": {key name from keyvault },
    "API_VERSION": "2016-10-01"

```

#### How can this be manually tested? (brief repro steps and corpnet-URL with change)
Deploy the code and create kevault in azure and try overriding the secret by adding in key vault.

#### Definition of done:
- [ ] Title of the pull request is clear and informative
- [ ] Add pull request hyperlink to relevant TFS items
- [ ] For large or complex change: schedule an in-person review session
- [ ] This change has appropriate test coverage
- [ ] Get at least two approvals

#### Reminders DURING merge
1. If you're merging from a short-term (feature) branch into a long-term branch (like dev, release, or master) then "Squash and merge" to keep our history clean.
1. If merging from two longterm branches (like cherry picks from upstream, dev to release, etc) then "Create merge commit" to preserve individual commits.

[//]: # ( fyi: This content was heavily inspired by )
[//]: # ( 1 Our team's policies and processes )
[//]: # ( 2 https://github.com/sprintly/sprint.ly-culture/blob/master/pr-template.md )
[//]: # ( 3 The book "The Checklist Manifesto: How to Get Things Right" by Atul Gawande )
[//]: # ( 4 https://github.com/Azure/azure-event-hubs/blob/master/.github/PULL_REQUEST_TEMPLATE.md )

### CODE REVIEW

File: lms/envs/azure_msi_key_vault.py [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+        # get access token with Azure MSI 
+        result = requests.get(request_uri, params=payload, headers={'Metadata': 'true'})
+        access_token = result.json()['access_token']
+
+        # get value from Azure keyvault using access token
+        headers_credentials = {'Authorization': 'Bearer' + ' ' + (access_token)}
+        response = requests.get(request_url, headers=headers_credentials)
+        secrets = json.loads(response.json()['value'])
+
+    except:
```
Comment (eltoncarr) [inline lms/envs/azure_msi_key_vault.py]: put in comments why we are swallowing the error

File: lms/envs/aws.py [hunk 2]
```diff
@@ -107,6 +109,18 @@
...
+##################### NON-SECURE ENV CONFIG FROM  AZURE KEY VAULT ########
+# Override non-secure env congif from azure keyvault
+if ENV_TOKENS.get('GET_SECRETS_FROM_AZURE_KEYVAULT',False):
+    request_uri = ENV_TOKENS.get('AZURE_MSI_REQUEST_URI',None)
+    payload = ENV_TOKENS.get('KEYVAULT_PAYLOAD',None)
+    key_vault_url = ENV_TOKENS.get('KEYVAULT_URL',None)
+    non_secure_key = ENV_TOKENS.get('LMS_NON_SECURE_KEY_NAME',None)
+    api_version = ENV_TOKENS.get('API_VERSION',None)
+    secrets = get_configs_from_keyvault(request_uri,payload,key_vault_url,non_secure_key,api_version)
+    for each_secret in secrets:
```
Comment (eltoncarr) [inline lms/envs/aws.py]: - [ ] Secret

File: lms/envs/aws.py [hunk 3]
```diff
@@ -465,6 +479,15 @@
 with open(CONFIG_ROOT / CONFIG_PREFIX + "auth.json") as auth_file:
     AUTH_TOKENS = json.load(auth_file)
 
+##################### SECURE AUTH FROM AZURE KEY VAULT ########
+# Override secure auth items from azure keyvault
+if ENV_TOKENS.get('GET_SECRETS_FROM_AZURE_KEYVAULT',False):
+    secure_keys = ENV_TOKENS.get('LMS_SECURE_KEY_NAME',None)
+    secrets = get_configs_from_keyvault(request_uri,payload,key_vault_url,secure_keys,api_version)
+    print(secrets)
```
Comment (eltoncarr) [inline lms/envs/aws.py]: - [ ] remove print

File: cms/envs/aws.py [hunk 4]
```diff
@@ -94,6 +96,18 @@
 with open(CONFIG_ROOT / CONFIG_PREFIX + "env.json") as env_file:
     ENV_TOKENS = json.load(env_file)
 
+##################### NON-SECURE ENV CONFIG FROM  AZURE KEY VAULT ########
+# Override non-secure env congif from azure keyvault
```
Comment (eltoncarr) [inline cms/envs/aws.py]: - [ ] congif = config

**Comment by manikarthikk** [comment]
@eltoncarr  @btelnes  @vivalams  
can you give a final review on this change.

**Pull request merged**